app/controllers/events_controller.rb

```diff
@@ -4,6 +4,17 @@ class EventsController < ApplicationController
   before_action :logged_in_user, only: [:new, :create]
   before_action :superadmin?, except:[:show, :index]
 
+  def index
+    upcoming = Event.upcoming(10)
+    query = params['q']
+    @events = query.blank? ? upcoming : upcoming.search(query)
```

Comment:
having `upcoming` as a receiver seems not right as the `search` method still searches the **whole database** and gives back **all records** matching the query